packages/@ama-sdk/schematics/schematics/typescript/core/index.ts

```diff
@@ -36,119 +66,189 @@ const getPathObjectTemplate = (pathObj: PathObject) => {
+  };
+};
+
 /**
  * Generate a typescript SDK source code base on swagger specification
  * @param options
  */
 function ngGenerateTypescriptSDKFn(options: NgGenerateTypescriptSDKCoreSchematicsSchema): Rule {
 
-  const specPath = path.resolve(process.cwd(), options.specPath);
-  const targetPath = options.directory || '';
-  const globalProperty = options.globalProperty;
+  return (tree, context) => {
+    const targetPath = options.directory || '';
+    const generatorOptions = getGeneratorOptions(tree, context, options);
 
-  const generateOperationFinder = async (): Promise<PathObject[]> => {
-    const swayOptions = {
-      definition: path.resolve(options.specPath)
+    /**
+     * rule to clear previous SDK generation
+     */
+    const clearGeneratedCode = () => {
+      treeGlob(tree, path.posix.join(targetPath, 'src', 'api', '**', '*.ts')).forEach((file) => tree.delete(file));
+      treeGlob(tree, path.posix.join(targetPath, 'src', 'models', 'base', '**', '!(index).ts')).forEach((file) => tree.delete(file));
+      treeGlob(tree, path.posix.join(targetPath, 'src', 'spec', '!(operation-adapter|index).ts')).forEach((file) => tree.delete(file));
+      return tree;
     };
-    const swayApi = await sway.create(swayOptions);
-    const extraction = swayApi.getPaths().map((obj) => ({
-      path: `${obj.path as string}`,
-      regexp: obj.regexp as RegExp,
-      operations: obj.getOperations().map((op: any) => {
-        const operation: Operation = {
-          method: `${op.method as string}`,
-          operationId: `${op.operationId as string}`
-        };
-        return operation;
-      }) as Operation[]
-    }));
-    return extraction || [];
-  };
 
-  /**
-   * rule to clear previous SDK generation
-   * @param tree
-   * @param _context
-   */
-  const clearGeneratedCode = (tree: Tree, _context: SchematicContext) => {
-    treeGlob(tree, path.posix.join(targetPath, 'src', 'api', '**', '*.ts')).forEach((file) => tree.delete(file));
-    treeGlob(tree, path.posix.join(targetPath, 'src', 'models', 'base', '**', '!(index).ts')).forEach((file) => tree.delete(file));
-    treeGlob(tree, path.posix.join(targetPath, 'src', 'spec', '!(operation-adapter|index).ts')).forEach((file) => tree.delete(file));
-    return tree;
-  };
+    const generateOperationFinder = async (): Promise<PathObject[]> => {
+      const swayOptions = {
+        definition: path.resolve(generatorOptions.specPath!)
+      };
+      const swayApi = await sway.create(swayOptions);
+      const extraction = swayApi.getPaths().map((obj) => ({
+        path: `${obj.path as string}`,
+        regexp: obj.regexp as RegExp,
+        operations: obj.getOperations().map((op: any) => {
+          const operation: Operation = {
+            method: `${op.method as string}`,
+            operationId: `${op.operationId as string}`
+          };
+          return operation;
+        }) as Operation[]
```

Comment:
```suggestion
        operations: obj.getOperations() as Operation[]
```
would not do the job?

Reply:
I have just tested this and the type casting doesn't seem to be working, so I updated the code to have a simpler mapping